.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,44 @@
+name: Build and Push Docker Image
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'apps/**'
+      - 'docker/explorer/**'
```

Comment:
1. push build only for branch main
2. check exactly what explorer depends on and build only for specific dir in apps, apps/explorer, apps/ui, etc...